test/integration/testExecute.js

```diff
@@ -213,14 +218,37 @@ describe('Execute test - variant', function ()
   var testCases =
     [
       {
-        name: 'raw xml',
+        name: 'xml_single_element',
+        type: 'XML',
+        fileExtension: '.xml',
+        sampleData: `<${TEST_HEADER}>${TEST_XML_VAL}</${TEST_HEADER}>`
+      },
+      {
+        name: 'xml_single_element_with_attribute',
+        type: 'XML',
+        fileExtension: '.xml',
+        sampleData: `<${TEST_HEADER} ${TEST_ATTRIBUTE_NAME}=${TEST_ATTRIBUTE_VALUE}>${TEST_XML_VAL}</${TEST_HEADER}>`,
+        ignoreAttributes: false
+      },
+      {
+        name: 'xml_with_attribute_and_custom_parser',
+        type: 'XML',
+        fileExtension: '.xml',
+        sampleData: `<node><${TEST_HEADER} ${TEST_ATTRIBUTE_NAME}=${TEST_ATTRIBUTE_VALUE}>${TEST_XML_VAL}</${TEST_HEADER}></node>`,
+        ignoreAttributes: false,
+        attributeNamePrefix: '##'
+      },
+      {
+        name: 'xml_skip_attributes',
         type: 'XML',
         fileExtension: '.xml',
+        sampleData: `<node><${TEST_HEADER} ${TEST_ATTRIBUTE_NAME}=${TEST_ATTRIBUTE_VALUE}>${TEST_XML_VAL}</${TEST_HEADER}></node>`
```

Comment:
what about test with xml nodes only with attributes but without any text?

Reply:
I'll add that case.